src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/CollectionExpressionTypeKind.cs

```diff
@@ -12,6 +12,7 @@ internal enum CollectionExpressionTypeKind
         ReadOnlySpan,
         CollectionBuilder,
         ImplementsIEnumerable,
+        ImplementsIEnumerableWithIndexer,
```

Comment:
My understanding is that `DictionaryInterface` and ` ImplementsIEnumerableWithIndexer` are the two cases added for dictionaries, is that right?
nit: Consider putting this at the end of the enum. That would keep the basic collection cases first, then the new dictionary case in second half (could be split with a comment)